packages/teleterm/src/preload.ts

```diff
@@ -16,12 +17,28 @@ const loggerService = createLoggerService({
 
 PreloadLogger.init(loggerService);
 
-const tshClient = createTshClient(runtimeSettings.tshd.networkAddr);
-const ptyServiceClient = createPtyService(runtimeSettings);
+contextBridge.exposeInMainWorld('electron', getElectronGlobals());
 
-contextBridge.exposeInMainWorld('electron', {
-  mainProcessClient,
-  tshClient,
-  ptyServiceClient,
-  loggerService,
-} as ElectronGlobals);
+async function getElectronGlobals(): Promise<ElectronGlobals> {
+  try {
+    const [addresses, credentials] = await Promise.all([
+      mainProcessClient.getResolvedChildProcessAddresses(),
+      getGrpcClientCredentials(runtimeSettings),
+    ]);
+    const tshClient = createTshClient(addresses.tsh, credentials.tsh);
+    const ptyServiceClient = createPtyService(
+      addresses.shared,
+      credentials.shared,
+      runtimeSettings
+    );
+
+    return {
+      mainProcessClient,
+      tshClient,
+      ptyServiceClient,
+      loggerService,
+    };
+  } catch (e) {
+    return e;
```

Comment:
Couldn't we remove this `catch` branch altogether and get rid of the error handling here?

https://github.com/gravitational/webapps/blob/344e62cf6f3143b6efebd2745dbbcc2bcea14dc1/packages/teleterm/src/ui/boot.tsx#L31-L37

AFAIR context bridge handles errors just fine, minus any custom properties which we don't need to worry about in this case anyway.

Reply:
Right, these error handlers are unnecessary here. I've just tested it, the app works the same without them.

I left the handler in the `boot` function to show the actual error instead of a blank screen.